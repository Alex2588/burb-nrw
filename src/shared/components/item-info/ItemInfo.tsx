import React from 'react';

import './item-info.styles.scss';

interface ItemInfo {
    name: string,
    image?: { medium: string, original: string },
    summary?: string
}

type ItemInfoProps<T> = { item: T };

const ItemInfoComponent = <T extends ItemInfo>({ item }: ItemInfoProps<T>) => {
    return (
        <div className="item-info">
            <div className="item-name">
                <h2>{item.name}</h2>
            </div>
            <div className="item-content">
                <div className="item-image">
                    <img src={item.image ? item.image.medium : `${process.env.PUBLIC_URL}/noimage.png`} alt=""/>
                </div>
                <div className="item-summary">
                    {item.summary ? item.summary.replace(/(<([^>]+)>)/gi, "") : `We don't have a summary for ${item.name} yet`}
                </div>
            </div>
        </div>
    );
}

export default ItemInfoComponent;