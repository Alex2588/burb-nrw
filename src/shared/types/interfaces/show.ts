export interface Show {
    id: string,
    url: string,
    name: string,
    type: string,
    language: string,
    genres: string[],
    status: string,
    runtime: number,
    premiered: string,
    officialSite: string,
    schedule: { time: string, days: string[] },
    rating: { average: number },
    weight: number,
    network: {
        id: number,
        name: string,
        country: {
            name: string,
            code: string,
            timezone: string
        }
    },
    externals: {
        tvrage: number,
        thetvdb: number,
        imdb: string
    },
    image: {
        medium: string,
        original: string
    },
    summary: string,
    updated: string,
    _links: {
        self: {
            href: string
        },
        previousepisode: {
            href: string
        }
    }
}