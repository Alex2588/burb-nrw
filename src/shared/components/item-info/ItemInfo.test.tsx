import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import { mock } from 'jest-mock-extended';

import ItemInfo from './ItemInfo';
import { Show } from 'shared/types/interfaces/show';

test('renders show info correctly if we pass show object as a prop', () => {
    const show = mock<Show>();
    show.name = 'Great show';
    show.summary = 'This is an amazing show';
    
    const { container } = render(<ItemInfo item={show} />, { wrapper: MemoryRouter });

    expect(screen.getByText(/Great show/i)).toBeInTheDocument();
    expect(screen.getByText(/This is an amazing show/i)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
});