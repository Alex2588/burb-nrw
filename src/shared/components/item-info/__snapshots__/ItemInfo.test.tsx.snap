// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders show info correctly if we pass show object as a prop 1`] = `
<div
  class="item-info"
>
  <div
    class="item-name"
  >
    <h2>
      Great show
    </h2>
  </div>
  <div
    class="item-content"
  >
    <div
      class="item-image"
    >
      <img
        alt=""
      />
    </div>
    <div
      class="item-summary"
    >
      This is an amazing show
    </div>
  </div>
</div>
`;
